src/lib/history.ts

```diff
@@ -0,0 +1,11 @@
+export default {
+  on: window.addEventListener.bind(window),
```

Comment:
Nit: Not a fan on the `on()` syntax. That’s a node idiom and something that we have avoided in this code base. I’d prefer either an explicit `addEventListener` or maybe even just a `addPopStateListener`.

Reply:
`addPopStateListener` sounds good, I'll use that.